wordpress-mobile/WordPress-iOS
# Remove WPMediaPicker dependency from Tenor picker

This PR reimplements Tenor (Free GIFs) picker without the `WPMediaPicker` framework.

I took some liberties with the design and tweaked it a little bit to match the recently redesigned screens.

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/a90af3c5-c064-4959-b235-21fb4ee2eaf7

## Changes

- Add `ExternalMediaPickerViewController` as a replacement for `TenorPicker` based on the `WPMediaPicker` framework
- Add `MediaPreviewController` modeled after `QLPreviewController`. I’m using `WPImageViewController` to display images, which is part of the app (not WPMediaPicker), so it won’t prevent us from removing it.
- Integrate the new picker in the following flows: Aztec and Gutenberg post editor, Site Media screen
- Update MediaImageService to support images from external sources. It takes advantage of all the improvements
- Extract `ImageDecoder` from `MediaImageService`
- Improve `ImageDownloader` by moving its work to the background, adding image decompression in the background
- Add `ImageView` to managing image downloads using the new architecture and design: solid gray placeholders, etc

## Known Issues

I think pagination in Tenor is not working, but it's a production issue. I'm not prioritizing fixing it because this screen is not used often.

## Testing

For every of the following flows:

- Add Image to Gutenberg
- Add Gallery to Gutenberg
- Add image to Aztec
- Add images to Site Media

Test the following:

- Verify that the new picker is displayed
- Verify that "Cancel" dismisses the screen and doesn't add any assets if the selection is made
- Verify that for multiple selection, the selection it oredered
- Verify that for multiple selection, a toolbar with a number of selected of items is displayed
- Verify that when you tap a toolbar button indicating the selection count the fullscreen gallery is displayed
- Verify that the fullscreen gallery renders GIFs with animation
- Verify that you can zoom in in the gallery
- Verify that when you tap "Done" the selected image are uploaded


## Regression Notes
1. Potential unintended areas of impact: Tenor (Aztec and Gutenberg post editor, Site Media screen)
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:


UI Changes testing checklist:
